x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/util/PlanStreamOutput.java

```diff
@@ -21,4 +24,13 @@ public interface PlanStreamOutput {
      * @throws IOException
      */
     boolean writeAttributeCacheHeader(Attribute attribute) throws IOException;
+
+    static void writeEsField(PlanStreamOutput output, EsField field) throws IOException {
+        output.writeEsField(field);
+    }
+
+    void writeEsField(EsField field) throws IOException;
+
+    // TODO get rid of this method when we unify esql and esql-core
+    <V> void writeMap(Map<String, V> map, Writeable.Writer<V> valueWriter) throws IOException;
```

Comment:
You can use `<S extends StreamInput & PlanStreamInput>` in the read method like `Source` does to avoid this.

Reply:
Probably it's easier than that, I just have to cast in writeEsField()